### PR TITLE
Today indication

### PR DESCRIPTION
**Pull request by mhib**
@lkudyba @pbzymek @s1mplex @quirk0o @adamzima @jtkpiotr

### CODE REVIEW

File: client/app/components/calendar/TimeIndicatorRow.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import moment from 'moment';
+import instanceOfMoment from 'proptypes/moment';
+import style from './calendar.scss';
+import { v4 as uuid } from 'node-uuid';
+
+export default class TimeIndicatorRow extends React.Component {
+  static propTypes = {
+    days:                     React.PropTypes.arrayOf(instanceOfMoment).isRequired,
+    unitEventLengthInSeconds: React.PropTypes.number.isRequired,
+    rowHeight:                React.PropTypes.string
```
Comment (adamzima) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: Please be consistent with indentation :D

File: client/app/components/calendar/TimeIndicatorRow.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    );
+  }
+
+  _isToday(day) {
+    return moment(day).isSame(moment(), 'day');
+  }
+
+  _topVal() {
+    const minutesFromMidnight = moment().diff(moment().clone().startOf('day'), 'minutes');
+    const top = minutesFromMidnight * parseInt(this.props.rowHeight, 10) / (this.props.unitEventLengthInSeconds / 60);
```
Comment (madejejej) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: What do you think about naming `parseInt(this.props.rowHeight, 10) / (this.props.unitEventLengthInSeconds / 60);` by extracting this into a variable?

File: client/app/components/calendar/TimeIndicatorRow.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  render() {
+    return (
+      <tr id="time-indicator-row"><td className="time-cell" />{this._timeIndicatorCells()}</tr>
+    );
+  }
+
+  _isToday(day) {
+    return moment(day).isSame(moment(), 'day');
+  }
```
Comment (quirk0o) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: Move to DateHelper?

File: client/app/components/calendar/TimeIndicatorRow.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  _topVal() {
+    const minutesFromMidnight = moment().diff(moment().clone().startOf('day'), 'minutes');
```
Comment (quirk0o) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: What is this wizardry?

Comment (quirk0o) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: Move to DateHelper? :P

Comment (quirk0o) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: No need to call `clone()` on `moment()` - `moment()` already creates a new instance of Moment

File: client/app/components/calendar/calendar.scss [hunk 5]
```diff
@@ -1,3 +1,11 @@
+$event-height: 2em;
+$event-font-size: 14px;
+$event-border: 1px;
+@function stripUnit($arg){
+  @return $arg/($arg * 0 + 1)
```
Comment (quirk0o) [inline client/app/components/calendar/calendar.scss:5]: What is this wizardry ? ^^

File: client/app/components/calendar/calendar.scss [hunk 6]
```diff
@@ -1,3 +1,11 @@
+$event-height: 2em;
+$event-font-size: 14px;
```
Comment (quirk0o) [inline client/app/components/calendar/calendar.scss:2]: Prefer `ems` over `px`

File: client/app/components/calendar/TimeIndicatorRow.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    return this.props.days.map(
+      (day, ind) => {
+        if (this._isToday(day)) {
+          return (
+            <td key={`IndicatorRow${ind}`} className="has-marker">
+              <div id="time-marker" style={{ top: this._topVal() }}></div>
+            </td>
+          );
+        }
+        return <td key={`IndicatorRow${ind}`} />;
```
Comment (quirk0o) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: Maybe you could use `react-if` here?

File: client/test/components/calendar/TimeIndicatorRow.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      expect(wrapper.find('td.has-marker')).to.have.exactly(1).descendants('div#time-marker');
+    });
+  });
+
+  describe('time-marker movement', () => {
+    it('updates top property', () => {
+      const topAttribute = () => wrapper.find('div#time-marker').prop('style').top;
+      const initialTop = parseInt(topAttribute(), 10);
+      clock.tick(SECONDS_IN_HOUR);
+      expect(parseInt(topAttribute(), 10) - initialTop).to.eq(rowHeight * 2);
```
Comment (quirk0o) [inline client/test/components/calendar/TimeIndicatorRow.spec.jsx]: Replace with `style.rowHeight`

File: client/test/components/calendar/TimeIndicatorRow.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+import { mount } from 'enzyme';
+// import TimeIndicatorRow from 'components/calendar/TimeIndicatorRow';
+
+const TimeIndicatorRow = proxyquire('../../../app/components/calendar/TimeIndicatorRow', {
+  './calendar.scss': { rowHeight: '32px' }
+}).default;
+
+describe('<TimeIndicatorRow />', () => {
+  const unitEventLength = 30 * 60;
+  const rowHeight = 32;
```
Comment (quirk0o) [inline client/test/components/calendar/TimeIndicatorRow.spec.jsx]: No longer needed

File: client/test/components/calendar/TimeIndicatorRow.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  let clock;
+  let days;
+  let wrapper;
+  const SECONDS_IN_HOUR = 1000 * 60 * 60;
+
+  before(() => {
+    clock = sinon.useFakeTimers();
+    days = [moment().clone().subtract(1, 'days'), moment(), moment().clone().add(1, 'days')];
```
Comment (quirk0o) [inline client/test/components/calendar/TimeIndicatorRow.spec.jsx]: No need to clone

File: client/test/components/calendar/TimeIndicatorRow.spec.jsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import moment from 'moment';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+// import TimeIndicatorRow from 'components/calendar/TimeIndicatorRow';
```
Comment (quirk0o) [inline client/test/components/calendar/TimeIndicatorRow.spec.jsx]: :)

File: client/app/components/calendar/calendar.scss (hunk 5 above)
Comment (mhib) [inline client/app/components/calendar/calendar.scss:5]: Function's name explains it pretty well.

File: client/app/components/calendar/calendar.scss (hunk 6 above)
Comment (mhib) [inline client/app/components/calendar/calendar.scss:2]: We need to convert em to px, so no.

File: client/app/components/calendar/TimeIndicatorRow.jsx (hunk 7 above)
Comment (adamzima) [inline client/app/components/calendar/TimeIndicatorRow.jsx]: :+1:

File: client/app/components/calendar/calendar.scss (hunk 5 above)
Comment (adamzima) [inline client/app/components/calendar/calendar.scss:5]: I still don't get it - `$arg*0 + 1` - why not `1` :D?

Comment (lkudyba) [inline client/app/components/calendar/calendar.scss:5]: https://css-tricks.com/snippets/sass/strip-unit-function/

**Comment by adamzima** [comment]
@mhib @lkudyba I'm so proud of you!

**Pull request merged**